src/components/CountryDetails.js

```diff
@@ -0,0 +1,47 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { NavLink, useParams } from "react-router-dom";
+
+
+function CounrtyDetails(props) {
+    
+    const { countryId } = useParams();
+    const [details, setDetails] = useState(null);
+
+    useEffect(() => {
+        axios.get(process.env.REACT_APP_API_BASE_URL+ '/' +countryId)
+        .then(response => {
+            setDetails(response.data);
+        })
+        .catch((e) => {
+            console.log(e);
+        });
+        // with json
+        // const aCountry = props.countries.find(element => element.alpha3Code === countryId)
+        // setDetails(aCountry);
+    }, [countryId, props.countries])
```

Comment:
Here we dont need to put props.countries, we get this data at the first render of app.js . countryId is enough